### DISCUSSION TITLE
How to add extra field on the schema?

### DISCUSSION THREAD

**Discussion by MdSamsuzzohaShayon** [fastapi-users/fastapi-users#1039] [Q&A]
Hello everyone, 
I am new here,
I am using this example __https://fastapi-users.github.io/fastapi-users/10.1/configuration/full-example/__

```
class UserCreate(schemas.BaseUserCreate):
    first_name: str
    # name: str
```

I have added an extra field but it's giving me an error which is `TypeError: 'first_name' is an invalid keyword argument for User`

How can I solve this and add extra field?

**Comment by frankie567** [comment]
Hello @MdSamsuzzohaShayon 👋 

Adding you field to `UserCreate` is only a first step, but you need to add this field in several other places.

## Schemas

You need to add it to `UserCreate`, as you did, but also to `UserRead` (so that `first_name` is output in the API) and `UserUpdate` (so that the user can update its `first_name`).

```py

class UserRead(schemas.BaseUser[uuid.UUID]):
    first_name: str


class UserCreate(schemas.BaseUserCreate):
    first_name: str


class UserUpdate(schemas.BaseUserUpdate):
    first_name: Optional[str]  # Optional because the user can do a partial update
```

## Model

Most importantly, you need to tell your SQL database about this new column. So you need to define a new proper SQLAlchemy column so the data can be stored in the DB.

```py
class User(SQLAlchemyBaseUserTableUUID, Base):
    first_name: str = Column(String(length=1024), nullable=False)
```

With this, it should work properly 👍

**Comment by youssefCeres** [comment]
Hello,

I have the same problem here is my schemas : 

```
import uuid
from typing import Optional
from fastapi_users import schemas
```

```
class UserRead(schemas.BaseUser[uuid.UUID]):
    first_name: Optional[str]
    last_name: Optional[str]
```


```
class UserCreate(schemas.BaseUserCreate):
    first_name: Optional[str]
    last_name: Optional[str]
```

```
class UserUpdate(schemas.BaseUserUpdate):
    first_name: Optional[str]
    last_name: Optional[str]
```


My model :

```
class User(SQLAlchemyBaseUserTableUUID, Base):
    oauth_accounts: List[OAuthAccount] = relationship("OAuthAccount", lazy="joined")
    first_name: Optional[str]
    last_name: Optional[str]
```


Here is the table in db :

```
                                      Table "public.user"
     Column      |          Type           | Modifiers | Storage  | Stats target | Description 
-----------------+-------------------------+-----------+----------+--------------+-------------
 email           | character varying(320)  | not null  | extended |              | 
 hashed_password | character varying(1024) | not null  | extended |              | 
 is_active       | boolean                 | not null  | plain    |              | 
 is_superuser    | boolean                 | not null  | plain    |              | 
 is_verified     | boolean                 | not null  | plain    |              | 
 id              | uuid                    | not null  | plain    |              | 
 first_name      | character varying(320)  |           | extended |              | 
 last_name       | character varying(320)  |           | extended |              | 

```

But when I try to create a user with all these fields, I still got :

`'first_name' is an invalid keyword argument for User`

Any hints plz ?

**Reply by frankie567** [reply]
The answer is in my previous message:

> ## Model
> Most importantly, you need to tell your SQL database about this new column. So you need to define a new proper SQLAlchemy column so the data can be stored in the DB.
> 
> ```python
> class User(SQLAlchemyBaseUserTableUUID, Base):
>     first_name: str = Column(String(length=1024), nullable=False)
> ```

**Reply by youssefCeres** [reply]
Thanks! @frankie567
